sound/soc/sof/intel/byt.c

```diff
@@ -265,17 +265,18 @@ static void byt_get_registers(struct snd_sof_dev *sdev,
 			      struct sof_ipc_panic_info *panic_info,
 			      u32 *stack, size_t stack_words)
 {
+	u32 offset = sdev->dsp_oops_offset;
+
 	/* first read regsisters */
-	sof_mailbox_read(sdev, sdev->dsp_oops_offset, xoops, sizeof(*xoops));
+	sof_mailbox_read(sdev, offset, xoops, sizeof(*xoops));
```

Comment:
this doesn't look right, you can't use sizeof on a structure with a variable-length array. Was is xoops->arch_hdr.totalsize as for bdw?

Reply:
@kv2019i sizeof struct with vararray is basiacally sizeof of struct with vararray of size 0. Real size of oops will be oops size + (before apl 32, later 64) * 4, probably there will be more different things in the future, that's why we have totalsize field in the oops header, it's the real size of whole struct (including headers and vararray).

Reply:
D'oh, copy'n'paste error, will fix.

Reply:
@plbossart sizeof will ignore "ar[]", so using sizeof() is ok here. 

To really dump the AR variables, that is a more complex change as then we'd need to change how the oops area is allocated (now users of this code allocate with sizeof so there's no space for AR register list).

We haven't support AR register dumping before, so I'd rather do this separately if needed.

Reply:
ok, then you should add a comment that this is intentional.

Reply:
@plbossart Ack, now uploaded a new patch that has an explicit comment (plus alignment to Janus'es newer patch).